radix-transactions/src/validation/transaction_validator.rs

```diff
@@ -577,6 +629,69 @@ impl TransactionValidator {
         })
     }
 
+    pub fn check_reference_limits(
+        &self,
+        subintent_references: Vec<&IndexSet<Reference>>,
+    ) -> Result<(), TransactionValidationError> {
+        let mut total = 0;
+        for (index, refs) in subintent_references.iter().enumerate() {
+            if refs.len() > self.config.max_references_per_intent {
+                return Err(TransactionValidationError::TooManyReferencesForIntent {
+                    index,
```

Comment:
From the transaction builder perspective, we don't have an "intent index" abstraction at all.

We either have "Transaction Intent" OR `SubintentIndex` into the non_root_subintents array (i.e. 0 = first non-root subintent). Feel free to rename `SubintentIndex` to `NonRootSubintentIndex` if you want, I think it got missed in the name refactor.

Reply:
In terms of this error - I'm happy if we just drop the index :)

Reply:
I think it's better to know which intent went wrong, for debugging purpose.

For partial transaction,  there is no "Transaction Intent". The index of intents is probably the easiest way to reveal such information.

Reply:
> I think it's better to know which intent went wrong, for debugging purpose.

Sure - but then I'd argue that we should map a `ManifestValidationError` into a `LocatedManifestValidationError` inside `TransactionValidationError::ManifestValidationError` with a location of either an `IntentLocation::Root` or `IntentLocation::NonRootSubintent(SubintentIndex)`, and could be done for all these validations.

Happy to do this when I add more tests.